assignments/array-methods.js

```diff
@@ -2,82 +2,556 @@
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
 let fullName = [];
+runners.forEach(runner =>
+  fullName.push(runner.first_name + " " + runner.last_name)
+);
+
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+let allCaps = runners.map(runner => runner.first_name.toUpperCase());
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter(runner => runner.shirt_size === "L");
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((acc, runner) => {
+  return acc + runner.donation;
+}, 0);
+
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+// Who are the top 10 with highest donations?
+let sortedDonations = runners.sort((a, b) =>
+  a.donation > b.donation ? -1 : 1
+);
+let largestDonations = sortedDonations.slice(0, 10);
+
+console.log(largestDonations);
 
 // Problem 2
+// What's the distribution of shirt sizes?
+let twoxlShirts = runners.filter(runner => runner.shirt_size === "2XL");
+let xlShirts = runners.filter(runner => runner.shirt_size === "XL");
+largeShirts = runners.filter(runner => runner.shirt_size === "L");
+let mediumShirts = runners.filter(runner => runner.shirt_size === "M");
+let smallShirts = runners.filter(runner => runner.shirt_size === "S");
+let xsShirts = runners.filter(runner => runner.shirt_size === "XS");
+console.log(twoxlShirts.length);
+console.log(xlShirts.length);
+console.log(largeShirts.length);
+console.log(mediumShirts.length);
+console.log(smallShirts.length);
+console.log(xsShirts.length);
+
+shirtSizes = {
+  XS: 0,
+  S: 0,
+  M: 0,
+  L: 0,
+  XL: 0,
+  "2XL": 0
+};
+
+let shirtCategories = runners.reduce((acc, runner) => {
+  let key = runner.shirt_size;
+  acc[key]++;
+  return acc;
+}, shirtSizes);
+
+console.log(shirtCategories);
+
```

Comment:
Again, this is amazing. Great job doing it manually with filters, and then realizing you can do it all in a single reduce. That is super impressive.